rust/opendp_tooling/src/codegen/python.rs

```diff
@@ -228,9 +230,20 @@ def {then_name}(
         String::new()
     };
 
+    let deprecated_decorator = func
+        .deprecation
+        .as_ref()
+        .map(|deprecation| {
+            format!(
+                "@deprecated(version=\"{}\", reason=\"{}\")\n",
+                deprecation.version, deprecation.note
```

Comment:
If "since" is not set, then the code will generate `@deprecated(version="", reason="xyz")`. I think it should either error or not include `version=""`. Similarly for reason.

Reply:
Good catch!

Reply:
hmm: I'm slightly confused. I've gone as far as adding a `panic!` above this line, and I don't see the error come through when I `cargo check`, although there are a lot of warning about unused variables.

If this is something we want to validate, maybe we should try to catch it sooner? Like having a `Deprecation::new()` that checks values when the struct is created?

Reply:
It would be better to throw a darling error back in `bootstrap/docstring.rs` when you first parse the deprecated attribute.
https://github.com/opendp/opendp/pull/1961/files#diff-54f929d780d7b93aface00e19d0f0016d7bdc78b0edf0ef3a4c04b2face20ca8R49-R50

Might look like this:
```rust
                    since: meta.since.ok_or_else(|| Error::custom("`since` must be specified").with_span(&attr))?,
                    note: meta.note.ok_or_else(|| Error::custom("`note` must be specified").with_span(&attr))?,
```

Reply:
If it compiles with `with_span`, the error will point at the attribute.

Reply:
I've made that change, but neither `cargo build --all-features` nor `cargo check --all-features` report an error with `since` deleted in one deprecation. Would I look somewhere else to see this error reported?